bokeh/models/glyph.py

```diff
@@ -75,7 +76,7 @@ def parameters(cls):
                 kind=Parameter.POSITIONAL_OR_KEYWORD,
 
                 # for positional arg properties, default=None means no default
-                default=Parameter.empty if default is None else default
+                default=Parameter.empty if default is None or is_field(default) else default
```

Comment:
This condition is a bit circumspect, I had to think about it for a bit. Maybe update the comment to explain that "args" that are explicitly specified as `field` are intended to be required in glyph methods?

Reply:
I changed this, so that arguments are processed from last to first. This way first non-default argument will make all following non-default, even if they have defaults (as it is now possible), thus avoiding gaps in positional arguments with and without defaults values.